Update packages to latest version

droberts-ctrlo opened the pull request:
Also update files where new ESLint errors were caught,
fix syntax where this has changed in test files,
and add new ESLint config

droberts-ctrlo committed:
Update packages to latest version

Also update files where new ESLint errors were caught,
and fix syntax where this has changed in test files

daurnimator commented on eslint.config.mjs:
```diff
@@ -0,0 +1,37 @@
+import { defineConfig } from "eslint/config";
+
+export default defineConfig([
+	{ files: ["**/*.{js,mjs,cjs}"], languageOptions: { globals: {...globals.browser, ...globals.node, ...globals.worker, ...globals.jest} } },
```
Can we turn `jest` on *only* inside of the test folder as the old config does?

daurnimator reviewed (commented):
> fix syntax where this has changed in test files,

I don't see that in the PR?

> and add new ESLint config

It indeed looks like there is a new config format (https://eslint.org/docs/latest/use/configure/configuration-files vs https://eslint.org/docs/latest/use/configure/configuration-files-deprecated).
Thanks!

droberts-ctrlo committed:
Changes as suggested in review

Update ci workflow with currently supported NodeJS version
Add variable to catch statement
Add appropriate eslint-disable directive
Update ESLint config
Add Node 20 to CI script
Updated CI script versions

daurnimator reviewed (approved):
Thankyou!
I'll squash merge this now.
In future I'd love it if you could amend/fixup your commits as you go, rather than pushing new ones at the end in response to review

I find the script at https://github.com/hashbang/dotfiles/blob/9d4ffd7e33814df1dfabcd48aca26a30ac6ee6f2/git/.local/bin/git-fixup#L1 to be very useful for working in git.

daurnimator commented:
Merged!
via 31a10463d29e91228da2b12f833efa3ceae908c2
